File: semester2/2/programming/lab3/task1.cpp
#include <iostream>

using namespace std;

int main() {
    signed long value, result = 1;
    short userChoice;
    cout << "Enter value: ";
    cin >> value;
	cout << "1. while (pre-conditional)" << endl;
	cout << "2. do-while (post-conditional)" << endl;
	cin << userChoice;
    if ((value & 1) == 0) value--;
	switch (userChoice) {
		case 1:
			while (value >= 1) {
				result *= value;
				value -= 2;
			}
			break;
		case 2:
			do {
				result *= value;
				value -= 2;
			} while (value >= 1);
			break;
	}
    cout << result << endl;
    return 0;
}

File: semester2/2/programming/lab3/task1.pas
Program lab3_1;
Var value, res:longint;
    userChoice:byte;
Begin
  res := 1;
  writeln('1. while (pre-conditional)');
  writeln('2. do-while (post-conditional)');
  read(userChoice);
  write('Enter value: ');
  read(value);
  if NOT(odd(value)) then value := value - 1;
  case (userChoice) of
    1:
      while (value >= 1) do begin
        res := res * value;
        value -= 2;
      end;
    2:
      repeat
        res := res * value;
        value -= 2;
      until (value <= 1);
  end;
  writeln(res);
End.

